File: src/main/java/coza/trojanc/receipt/printer/Printer.java
package coza.trojanc.receipt.printer;

import coza.trojanc.receipt.format.PrintFormatBuilder;

/**
 * @author Charl Thiem
 */
public interface Printer {

	/**
	 * Gets the layout builder instance for the printer
	 *
	 * @return layout builder
	 */
	PrintFormatBuilder getLayoutBuilder();

	/**
	 * Prints the layout that has been setup
	 */
	void print();
}

File: receipt-shared/src/main/kotlin/coza/trojanc/receipt/printer/Printer.kt
package coza.trojanc.receipt.printer

import coza.trojanc.receipt.format.PrintFormatBuilder

/**
 * @author Charl Thiem
 */
interface Printer {

    /**
     * Gets the layout builder instance for the printer
     *
     * @return layout builder
     */
    fun getLayoutBuilder(): PrintFormatBuilder

    /**
     * Prints the layout that has been setup
     */
    fun print()
}
